### ISSUE TITLE
Consider not using external images

### ISSUE THREAD

**Issue by dandv** [LunarLogic/starability#17]
Images can be [embedded using base64 encoding](http://stackoverflow.com/questions/1207190/embedding-base64-images), replaced with [inline SVG](https://css-tricks.com/icon-fonts-vs-svg/), or with simple [Unicode star characters](https://css-tricks.com/star-ratings/).

**Comment by StudioJunkyard** [comment]
I used the Unicode for my implementation. Works great.

**Comment by dandv** [comment]
@StudioJunkyard: does your version happen to be open source?

**Comment by StudioJunkyard** [comment]
It's within the styling for LibreCMS, but I can give you the CSS
````css
.rat{
 display:inline-block;
 unicode-bidi:bidi-override;
 direction:rtl
}
.rat > span{
 position:relative;
 width:1em;
 text-shadow:0 1px 1px #000
}
.rat > span.set,
.rat > span.set ~ span{
 color:transparent
}
.rat > span.set:after,
.rat > span.set ~ span:after{
 content:"\2605";
 position:absolute;
 left:0;
 color:gold
}
.rating{
 position:relative;
 float:left
}
.rating:not(:checked) > input{
 position:absolute;
 top:-9999px;
 clip:rect(0,0,0,0)
}
.rating:not(:checked) > label{
 float:right;
 width:1em;
 padding:0 .1em;
 overflow:hidden;
 white-space:nowrap;
 cursor:pointer;
 font-size:150%;
 line-height:1.2;
 color:#ddd;
 text-shadow:1px 1px #bbb, 2px 2px #666,.1em .1em .2em rgba(0,0,0,.5)
}
.rating:not(:checked) > label:before{
 content:'★ '
}
.rating > input:checked ~ label{
 color:#f70;
 text-shadow:1px 1px #c60,2px 2px #940,.1em .1em .2em rgba(0,0,0,.5)
}
.rating:not(:checked) > label:hover,
.rating:not(:checked) > label:hover ~ label{
 color:gold;
 text-shadow:1px 1px goldenrod,2px 2px #B57340,.1em .1em .2em rgba(0,0,0,.5)
}
.rating > input:checked + label:hover,
.rating > input:checked + label:hover ~ label,
.rating > input:checked ~ label:hover,
.rating > input:checked ~ label:hover ~ label,
.rating > label:hover ~ input:checked ~ label{
 color:#ea0;
 text-shadow:1px 1px goldenrod,2px 2px #B57340,.1em .1em .2em rgba(0,0,0,.5)
}
.rating > label:active{
 position:relative;
 top:2px;
 left:2px
}
.rating > label:after{
 content:attr(data-title);
 position:absolute;
 display:block;
 top:auto;
 bottom:30px;
 width:5px;
 height:auto;
 padding-left:7px;
 color:#000;
 font-size:10px;
 font-weight:400;
 text-shadow:0 0 0;
 word-wrap:break-word
}
````

and the HTML for displaying with the interaction:
````html
<span class="rat">
  <span<?php if($rr['cid']==5)echo' class="set"';?>>☆</span>
  <span<?php if($rr['cid']==4)echo' class="set"';?>>☆</span>
  <span<?php if($rr['cid']==3)echo' class="set"';?>>☆</span>
  <span<?php if($rr['cid']==2)echo' class="set"';?>>☆</span>
  <span<?php if($rr['cid']==1)echo' class="set"';?>>☆</span>
</span>
````
That should give you an idea.

**Comment by szynszyliszys** [comment]
@dandv @StudioJunkyard I've changed external images to base64 encoded sprites. I tried changing sprites to separate encoded star images, but it could cause problems with backwards compatibility for people with starability-slot animation in use. It also added way too many kB of code. Thank you for your feedback and help. :)

Why base64? Inline SVG seemed hard to implement here, as I take advantage of pseudoelements. I like the look of the star graphics used in starability and Unicode characters are a bit different.
